host-components/src/keep-loader.rs

```diff
@@ -0,0 +1,229 @@
+// SPDX-License-Identifier: Apache-2.0
+#[macro_use]
+extern crate serde_derive;
+
+use ::host_components::*;
+use std::io::prelude::*;
+use std::os::unix::net::{UnixListener, UnixStream};
+use std::process::Command;
+use std::sync::{Arc, Mutex};
+use std::thread;
+
+//IMPORTANT - after every change to this file, the following steps need to be run:
+// remove link file in /etc/systemd/user/
+// recreate link file in /etc/systemd/user/
+// `systemctl daemon-reload`
+/*
+pub const KEEP_INFO_COMMAND: &str = "keep-info";
+pub const KEEP_APP_LOADER_START_COMMAND: &str = "apploader-start"; // requires app_loader_bind_port to be provided
+pub const KEEP_LOADER_STATE_UNDEF: u8 = 0;
+pub const KEEP_LOADER_STATE_LISTENING: u8 = 1;
+pub const KEEP_LOADER_STATE_STARTED: u8 = 2;
+pub const KEEP_LOADER_STATE_COMPLETE: u8 = 3;
+pub const KEEP_LOADER_STATE_ERROR: u8 = 15;
+*/
+//TODO - put in a shared space
+#[derive(Serialize, Deserialize)]
+struct JsonCommand {
+    commandtype: String,
+    commandcontents: String,
+}
+
+fn main() {
+    println!("Welcome to a new keep-loader");
+
+    //get and parse args
+    let args: Vec<String> = std::env::args().collect();
+    //bind to unix socket
+    //await commands
+    //TODO - remove hard-coding!
+    println!("Keep-loader has received {} args", args.len());
+    let kuuid = args[1].clone();
+    println!("kuuid = {}", kuuid);
+    let bind_socket = format!("/tmp/enarx-keep-{}.sock", kuuid);
```

Comment:
You should really use `/var/run/user/$UID/enarx` instead of `/tmp`. Be a shiny example to others and don't use insecure directories.